pkgs/development/python-modules/jep/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, python, fetchPypi, buildPythonPackage, numpy, jre, numpySupport ? true }:
+
+let
+  inherit (lib) optional optionals;
+in
+
+buildPythonPackage rec {
+  pname = "jep";
+  version = "3.9.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "fM0lqSoZo5FQTnZpQBkL3NTWs6hIjKSjrcjrjNWBwMs=";
+  };
+
+  buildInputs = [ jre ];
+  propagatedBuildInputs = optionals numpySupport [ numpy ];
+
+  preCheck = optional numpySupport ''
+    export PYTHONHOME="${python.withPackages(_: [ numpy ])}"
+  '';
+
+  postFixup = ''
```

Comment:
Can't `makeWrapper` be used for this?

Reply:
I'm not sure.
This thing ends up pointing to the lib directory of the python environment we use as an input https://github.com/ninia/jep/blob/master/src/main/scripts/jep#L15. Instead we need it to point at the lib directory of the current derivation.
I think because of this we will need to rewrite / patch the script anyway, in which case I think this is simpler.